crates/binjs_io/src/entropy/baseline.rs

```diff
@@ -0,0 +1,229 @@
+            root = spec.get_root_name().to_str(),
+            depth = depth);
+
+        debug!(target: "baseline", "Roots: [{:?}]\n",
+            spec.resolved_sums_of_interfaces_by_name()
+                .get(spec.get_root_name())
+                .unwrap()
+                .iter()
+                .map(|node_name| InterfaceName::from_rc_string(node_name.to_rc_string().clone()))
+                .format(", "));
+
+        BaselineDictionaryBuilder {
+            spec,
+            null_name,
+            dictionary: Dictionary::new(depth, 0),
+        }
+    }
+
+    fn seed_one_field(
+        &mut self,
+        path: &IOPath,
+        type_spec: &spec::TypeSpec,
+        or_null: bool,
+        one_level_paths_by_interface_name: &mut HashMap<InterfaceName, HashSet<IOPath>>,
+    ) {
+        use self::spec::NamedType;
+        use self::spec::TypeSpec;
+        match type_spec {
+            TypeSpec::Array { ref contents, .. } => {
+                // Entering an array doesn't affect the path.
+                self.seed_one_field(
+                    path,
+                    contents.spec(),
+                    contents.is_optional(),
+                    one_level_paths_by_interface_name,
+                );
+            }
+            TypeSpec::TypeSum(ref sum) => {
+                for spec in sum.types() {
+                    self.seed_one_field(path, spec, or_null, one_level_paths_by_interface_name);
+                }
+            }
+            TypeSpec::NamedType(ref name) => {
+                let named_type = self.spec.get_type_by_name(name).unwrap(); // Grammar has been verified already.
+                match named_type {
+                    NamedType::Interface(ref interface) => {
+                        let interface_name =
+                            InterfaceName::from_rc_string(interface.name().to_rc_string().clone());
+
+                        // Store path => interface and interface => path.
+                        self.dictionary
+                            .interface_name_by_path
+                            .add_if_absent(path.borrow(), interface_name.clone());
+                        one_level_paths_by_interface_name
+                            .entry(interface_name.clone())
+                            .or_insert_with(|| HashSet::new())
+                            .insert(path.clone());
+                        if or_null {
+                            self.dictionary
+                                .interface_name_by_path
+                                .add_if_absent(path.borrow(), self.null_name.clone());
+                            one_level_paths_by_interface_name
+                                .entry(self.null_name.clone())
+                                .or_insert_with(|| HashSet::new())
+                                .insert(path.clone());
+                        }
+                    }
+                    NamedType::Typedef(ref def) => self.seed_one_field(
```

Comment:
can you use block here, to make this more similar to other block, for readability?